docs/sketch-specification.md

```diff
@@ -35,7 +35,7 @@ The following extensions are supported:
 - .cpp - C++ files.
 - .c - C Files.
 - .S - Assembly language files.
-- .h - Header files.
+- .h, .tpp, .ipp - Header files.
```

Comment:
```suggestion
- .h - Header files.
- .tpp, .ipp - Header files (available from Arduino CLI 0.18.4).
```
I think it will be important to specify at which version this was introduced in each of the official development tools, but I'm open to other ideas for how to format the information.

As I mentioned before, I'm not very happy with this approach and would like to establish a less messy and haphazard system, but it is the established practice in the specifications for the most part so for now I guess we're best to just stick with it.

One of the failings of this approach is that we don't necessarily know that the next release version will be 0.18.4.

Reply:
I don't think this is necessary, we already have the version selector on top for this. Also the next version will probably be a `0.19.0` since it will be breaking.
How do we handle the IDE version? Will we need to update the docs when we release the version of the IDE that includes the CLI with this change?

Reply:
>we already have the version selector on top for this

It makes sense in theory, but the reality is that the specification is usually updated as an afterthought long after the change to the code. So that's not a reliable indicator of when the change was made.

>Will we need to update the docs when we release the version of the IDE that includes the CLI with this change?

Yes, and for each of the tools: classic IDE, IDE 2.x, Arduino Web Editor.

Reply:
This is not just documenting the version, but also which tools have the capability. If there is no qualification, the reader would expect the feature is available in all official tools.

Reply:
Mmmh ok, I don't think it's the best approach but it's fine for me. I'll change it to `0.19.0` though.

Reply:
>I don't think it's the best approach

I agree completely. As I mentioned in our previous discussion, I think the right way is to version the specifications as a whole, with each release of a tool implementing a specific specification version.

But that approach of the specification driving the capabilities provided by the tools is a complete reversal of the current approach, and so likely to be challenging to accomplish.